ignite/distributed/utils.py

```diff
@@ -350,29 +351,80 @@ def all_reduce(
     return _model.all_reduce(tensor, op, group=group)
 
 
+def _all_gather_tensors_with_shapes(
```

Comment:
Same for this change, this is another feature, let's land it in a separate PR

Reply:
https://github.com/pytorch/ignite/pull/2947